### ISSUE TITLE
BUG with tinymce

### ISSUE THREAD

**Issue by batata004** [tinymce/tinymce#4492]
How to make TINYMCE stop removing empty span tags? On 3.x version it was pretty easy however on 4.x it's impossible. If tinymce finds an element like <span class="xx"></span> it simply removes this element.

Even using valid_elements with "#span[*]" to allow padding DOES NOT work! Tinymce wont padd and will remove the span anyway if it's empty.

You should create some method which allow users to stop tinymce cleaning process, it's doing a terrible job removing empty tags.

**Comment by SimonFc** [comment]
Hi,
Using `valid_elements`/`extends_valid_elements` and then inserting an empty span tag through the code plugin is working for me. A span is inserted with a non-breaking space character. Is there a special scenario where it wont work?

**Comment by batata004** [comment]
You used BOTH valid_elements/extends_valid_elements with the same value?

Check this -> http://fiddle.tinymce.com/lvgaab/5

I setted the valid_elements to make span tags be padded #. Click the rightmost of the XXX and press backspace 3 times. On the third time the SPAN should be padded BUT it does not happen and the span gets removed anyway.

**Comment by SimonFc** [comment]
Just one of them at a time, and that made it possible for me to insert empty span tags with the code plugin.

The behavior with the spans when you're erasing is expected though. The `valid_elements` option will make(in this case) span tags valid when you're getting the content from the editor. The behavior of removing empty span tags when when the content has been erased will remain the same.

**Comment by batata004** [comment]
@SimonFc I disagree with you (2nd paragraph). When you are editing a content (typing) and you configured valid_elements as `#div[*]` it pads the `div`s correctly are "runtime". This does not happen with span. It only works at initialization of the editor. So what is the point?? If it only works at initialization I could very well do a regex with PHP to pad the `span` before placing them on the editor.

Also, MS WORD, BR OFFICE WORD... all editors works the way I told you: you can delete the XXX and start typing anything else and you wont lose format/stlye.

**Comment by batata004** [comment]
@SimonFc THANKS A LOT!!! Do you know how many times I reported this bug? At least 3 times at github and at least 5 times at the "tinymce community" forum. There are tons of other people complaning about the same thing online but it looks like nobody at tinymce thought it was necessary.

Also, on 3.x version this bug was easily fixeable cause I could easily disable cleanup of the tinymce algorithm and enable MY OWN clean up! With 4.x version it's completely impossible to stop cleanup, and even worse, there is no way I can attach a custom cleanup function to replace the buggy cleanup function that comes wtih tinymce.

Anyway, I am glad you could take a look into this! I am really happy now! It's been a loooong time since I  have to deal with this bug. It's so annowing that some of my customers I had to keep the 3.x version ONLY because of this bug!

Good luck on finding a solution, I am here to help in anything you need!

**Comment by SimonFc** [comment]
We want to fix as many bugs as possible altough there're many things that will be given priority. I'll keep this bug in mind though. You are of course always welcome to open up a PR and contribute to the project if you were to find a solution!

**Comment by batata004** [comment]
@SimonFc  I really appreciate your kindness and I am pretty sure the community too! I would like to help fixing this bug (I really do) but I have no idea how Tinymce library works and I have very low JS background. If it were some problem with the file upload manager or something related to PHP I would be more than glad to help! 

Despite my lack of help in this project, I am really thankful for your attention into this issue!

**Comment by griffind** [comment]
@batata004 This seems related to other multiple tickets that have been filed with Tiny where you change the font family to something other than the default, type characters, delete all those characters, and since the span is deleted you are back to the default font. 

If you are typing in an editor, you expect the font family never to change unless you explicitly do so, or are editing within a tag with a different font family.

According to fiddle it seems that this became a problem in 4.6, since I dont see this issue in prior versions. Seen many tickets open in regards to this that were either closed or closed and also said to be fixed. I am surprised that these issues have seen no love, since in my opinion this is a ground level editor expected behavior.

**Comment by batata004** [comment]
@griffind and you know what? This exactly behaviour you reported (deleting 2 chars...) works perfectly on Google Docs. I am saying this because in another ticket I opened another user said that WHAT I want (and it looks like, what you also want) is not a "good behaviour". But you know what? Google Docs work exactly same way: delete 2 chars and the font/style does not get lost.

This is better explained in another thread that I already opened at https://github.com/tinymce/tinymce/issues/5741 but the guy said "There is no fix on the horizon as I say". Unbeliveable! A serious bug, that impacts a lot the practical usage of tinymce, gets dismissed like that.

**Comment by griffind** [comment]
@batata004 Yea it's the same. I've seen that issue, it's also the same with [#3309](https://github.com/tinymce/tinymce/issues/3309). 

Surprising that the current functionality is seen as a convenience. Ask any user to open up TinyMCE, and without typing anything, change the font from the default, and hit backspace. Now ask them what they would expect to happen. Most would say nothing and they would be wrong...their font selection has changed. This is not the behavior in any text editor that I am aware of.

**Comment by batata004** [comment]
@griffind exactly, I cant believe the TinyMCE devs thought it was a convenience... if at least there was a way to extend some property of tinymce and override it so this behaviour gets changed... Anyway, if anyone has any idea, I would be glad :)

**Comment by batata004** [comment]
comment

**Comment by baky30** [comment]
Still have that bug
